### ISSUE TITLE
Openhab 4.1 starts mass of SocketListener and JmDNS Threads in Docker

### ISSUE THREAD

**Issue by scriptwriter13** [openhab/openhab-core#3976]
The Version 4.1.0 generates a lot of SocketListener and JmDNS Processes which makes CPU high and available Memory low. This behavior is if you use it in docker in host network-mode (which you need if you want to use upnp-devices) and you have other microservices running which each has own networks which interconnecting their containers and visible by  openhab in hostmode. The CPU-usage is so high that GUI is not really usable any more.

## Expected Behavior
It should never start JmDNS and SocketListener on all existing network-interfaces which could be a mass of interfaces existing in docker, because normally usage of docker is that you host multiple microservices on a docker host.

## Current Behavior

Count the instances of Socketlistener and JmDNS
```
docker exec -it openhab-openhab-1 ps -T -p 26|grep Socket|wc -l
109
docker exec -it openhab-openhab-1 ps -T -p 26|grep JmDNS|wc -l
309
```
GUI very slow, hard to use, mostly unusable because it is not reacting.

Additional Information: the Variables `-Dorg.jupnp.network.useInterfaces=eth0 -Dorg.jupnp.network.useAddresses=192.168.2.244` set via EXTRA_JAVA_OPTS seems to be ignored in 4.1 and were working in 4.0.X

## Possible Solution
it should be possible to define 1 interface which the java-process is listen on so it will never bind to other interfaces. The possibility to specify an interface in gui is not doing that.

## Steps to Reproduce (for Bugs)
- install a docker host
- deploy some microservices which interconnecting at least 1 container to another by own network 
- then deploy openhab 4.1.0 container in host-mode

## Context

Several Discussions on 
[https://community.openhab.org/t/unsuccess-openhab-updated-from-4-0-2-to-4-1-0-on-docker-openhab-unresponsive/152282](https://community.openhab.org/t/unsuccess-openhab-updated-from-4-0-2-to-4-1-0-on-docker-openhab-unresponsive/152282)

Together we found out that there is a new feature existing in 4.1.0 which is named "Suggestionfinder". To see if is in relation with this we switched it off by addons.cfg:
```
suggestionFinderMdns = false
suggestionFinderUpnp = false
suggestionFinderIp = false
```
But problem is still existing so it could have another reason.

The UPNP-Binding is uninstalled so a relation from that is not possible.

Downgrade to 4.0.3 is possible and then problem is fully gone.

Problem is fully reproducable on upgrade to 4.1.0.

The problem is existing on other installations with docker too, see comments of the upper linked discussion in community

## Your Environment
Debian 11
Docker version 24.0.7
Raspi 4 8GB
55 virtual networks from docker existing today

(problem exists on other people which using 8 Core Intel CPU too)

**Comment by wborn** [comment]
I think this is a regression of https://github.com/openhab/openhab-core/pull/3906.

It looks like with this change it no longer respects  the "Network Configuration" settings like disabling IPv6 and only using one IP per interface to prevent these performance issues (https://github.com/eclipse-archived/smarthome/pull/6005).

**Comment by rkoshak** [comment]
> it should be possible to define 1 interface which the java-process is listen on so it will never bind to other interfaces. The possibility to specify an interface in gui is not doing that.

Assuming this is a regression like @wborn indicates, in the mean time you can set the following environment variable when starting the container:

    EXTRA_JAVA_OPTS="-Dorg.jupnp.network.useInterfaces=eno1"

There's more detail at https://deploy-preview-2190--openhab-docs-preview.netlify.app/docs/installation/docker.html#universal-plug-and-play-upnp which is a proposed update to the docs for Sonos speakers which have a similar problem.

**Comment by scriptwriter13** [comment]
> Assuming this is a regression like @wborn indicates, in the mean time you can set the following environment variable when starting the container:
> 
> ```
> EXTRA_JAVA_OPTS="-Dorg.jupnp.network.useInterfaces=eno1"
> ```

Hi @rkoshak,
unfortunatly i have since longer time this Variable in there. Unfortunatly ithe behavior exists and this Variable makes no difference

**Comment by pgfeller** [comment]
Hi @scriptwriter13,

```
                           _   _     _     ____  
   ___   ___   ___   ___  | | | |   / \   | __ ) 
  / _ \ / _ \ / _ \ / _ \ | |_| |  / _ \  |  _ \ 
 | (_) | (_) |  __/| | | ||  _  | / ___ \ | |_) )
  \___/|  __/ \___/|_| |_||_| |_|/_/   \_\|____/ 
       |_|       4.1.0 - Release Build

```

I use a openHAB 4.1.0 setup created with docker compose file, and here the parameter seems to work:

```log
42 results - 1 file

userdata/logs/upnp.log:
   61  2023-12-28 20:17:27.138 [DEBUG] [org.jupnp.transport.Router          ] - Starting networking services...
   62: 2023-12-28 20:17:27.141 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Analyzing network interface: veth1561349
   63: 2023-12-28 20:17:27.147 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Skipping unwanted network interface (-D org.jupnp.network.useInterfaces ): veth1561349
   64: 2023-12-28 20:17:27.147 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Ignoring non-usable network interface: veth1561349
   65: 2023-12-28 20:17:27.149 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Analyzing network interface: vetha10e96b
   66: 2023-12-28 20:17:27.150 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Skipping unwanted network interface (-D org.jupnp.network.useInterfaces ): vetha10e96b
   67: 2023-12-28 20:17:27.150 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Ignoring non-usable network interface: vetha10e96b
   68: 2023-12-28 20:17:27.150 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Analyzing network interface: vethdcbbcd8
   69: 2023-12-28 20:17:27.151 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Skipping unwanted network interface (-D org.jupnp.network.useInterfaces ): vethdcbbcd8
   70: 2023-12-28 20:17:27.151 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Ignoring non-usable network interface: vethdcbbcd8
   71: 2023-12-28 20:17:27.151 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Analyzing network interface: veth3fe1c68
   72: 2023-12-28 20:17:27.152 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Skipping unwanted network interface (-D org.jupnp.network.useInterfaces ): veth3fe1c68
   73: 2023-12-28 20:17:27.152 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Ignoring non-usable network interface: veth3fe1c68
   74: 2023-12-28 20:17:27.152 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Analyzing network interface: veth46c53ba
   75: 2023-12-28 20:17:27.153 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Skipping unwanted network interface (-D org.jupnp.network.useInterfaces ): veth46c53ba
   76: 2023-12-28 20:17:27.159 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Ignoring non-usable network interface: veth46c53ba
   77: 2023-12-28 20:17:27.159 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Analyzing network interface: vethce9c5ac
   78: 2023-12-28 20:17:27.161 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Skipping unwanted network interface (-D org.jupnp.network.useInterfaces ): vethce9c5ac
   79: 2023-12-28 20:17:27.161 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Ignoring non-usable network interface: vethce9c5ac
   80: 2023-12-28 20:17:27.168 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Analyzing network interface: vethfa0ffa3
   81: 2023-12-28 20:17:27.171 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Skipping unwanted network interface (-D org.jupnp.network.useInterfaces ): vethfa0ffa3
   82: 2023-12-28 20:17:27.171 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Ignoring non-usable network interface: vethfa0ffa3
   83: 2023-12-28 20:17:27.172 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Analyzing network interface: br-4a86aa3f87c6
   84: 2023-12-28 20:17:27.172 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Skipping unwanted network interface (-D org.jupnp.network.useInterfaces ): br-4a86aa3f87c6
   85: 2023-12-28 20:17:27.173 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Ignoring non-usable network interface: br-4a86aa3f87c6
   86: 2023-12-28 20:17:27.175 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Analyzing network interface: br-7406c5aa57f0
   87: 2023-12-28 20:17:27.176 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Skipping unwanted network interface (-D org.jupnp.network.useInterfaces ): br-7406c5aa57f0
   88: 2023-12-28 20:17:27.177 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Ignoring non-usable network interface: br-7406c5aa57f0
   89: 2023-12-28 20:17:27.180 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Analyzing network interface: docker0
   90: 2023-12-28 20:17:27.183 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Skipping network interface (down): docker0
   91: 2023-12-28 20:17:27.183 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Ignoring non-usable network interface: docker0
   92: 2023-12-28 20:17:27.184 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Analyzing network interface: eno1.4
   93: 2023-12-28 20:17:27.184 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Skipping unwanted network interface (-D org.jupnp.network.useInterfaces ): eno1.4
   94: 2023-12-28 20:17:27.185 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Ignoring non-usable network interface: eno1.4
   95: 2023-12-28 20:17:27.185 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Analyzing network interface: eno1
   96: 2023-12-28 20:17:27.186 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Discovered usable network interface: eno1
   97: 2023-12-28 20:17:27.186 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Analyzing network interface: lo
   98: 2023-12-28 20:17:27.186 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Skipping network interface (ignoring loopback): lo
   99: 2023-12-28 20:17:27.187 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Ignoring non-usable network interface: lo
  100: 2023-12-28 20:17:27.190 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Discovering addresses of interface: eno1
  103: 2023-12-28 20:17:27.199 [TRACE] [sport.impl.NetworkAddressFactoryImpl] - Discovered usable network interface address: 192.168.0.65
  104  2023-12-28 20:17:27.203 [DEBUG] [org.jupnp.transport.Router          ] - Init multicast receiver on interface: eno1
```

Please try the following setting to log the relevant data to verify if the interfaces are detected and ignored as expected:

```xml
<!-- UPNP -->
<Logger additivity="false" level="WARN" name="org.openhab.binding.sonos.internal.discovery"><AppenderRef ref="UPNP"/></Logger>
<Logger additivity="false" level="TRACE" name="org.openhab.core.net.NetUtil"><AppenderRef ref="UPNP"/></Logger>
<Logger additivity="false" level="WARN" name="org.openhab.core.io.transport.upnp.internal"><AppenderRef ref="UPNP"/></Logger>
<Logger additivity="false" level="WARN" name="org.openhab.core.config.discovery.upnp.internal.UpnpDiscoveryService"><AppenderRef ref="UPNP"/></Logger>
<Logger additivity="false" level="WARN" name="org.openhab.core.config.discovery.upnp"><AppenderRef ref="UPNP"/></Logger>
<Logger additivity="false" level="WARN" name="org.openhab.core.config.discovery"><AppenderRef ref="UPNP"/></Logger>
<Logger additivity="false" level="DEBUG" name="org.jupnp.transport.Router"><AppenderRef ref="UPNP"/></Logger>
<Logger additivity="false" level="TRACE" name="org.jupnp.transport.impl"><AppenderRef ref="UPNP"/></Logger>
```

I use a `.env` file with the following option:
```ini
EXTRA_JAVA_OPTS="-Duser.timezone=Europe/Berlin -Dorg.jupnp.network.useInterfaces=eno1"
```

In the compose file I also define the server IP - but I'm not sure if that is even necessary:
```yaml
  openhab:
    depends_on:
      - location
      - frontail
      - influxdb
      - grafana
      - zigbee2mqtt
    container_name: ${COMPOSE_PROJECT_NAME}-server
    image: pgfeller/openhab:4.1.0
    build:
      context: ../images/openhab
    restart: unless-stopped
    network_mode: host
    group_add:
      - tty
    volumes:
      - /etc/localtime:/etc/localtime
      - /etc/timezone:/etc/timezone
      - $_OPENHAB_CONF/ssh:/openhab/.ssh
      - $_OPENHAB_ADDONS:/openhab/addons
      - $_OPENHAB_CONF:/openhab/conf
      - $_OPENHAB_USERDATA:/openhab/userdata
    devices:
      - /dev/serial/by-id/usb-0658_0200-if00:/dev/ttyACM2
      - /dev/serial/by-id/usb-Prolific_Technology_Inc._USB-Serial_Controller_D-if00-port0:/dev/ttyUSB0
    environment:
      - CRYPTO_POLICY=unlimited
      - SSHPASS=$SSHPASS
      - EXTRA_JAVA_OPTS=${EXTRA_JAVA_OPTS} 
      - OPENHAB_HTTP_ADDRESS=192.168.0.65
```

Note: I customize the 4.1 image a bit - but this will not have an influence on this topic. Interesting is, what the logs say ...

**Comment by lolodomo** [comment]
> I think this is a regression of #3906.
> 
> It looks like with this change it no longer respects the "Network Configuration" settings like disabling IPv6 and only using one IP per interface to prevent these performance issues

@wborn : I see what you mean. I will try to propose a fix.

**Comment by pgfeller** [comment]
@lolodomo 

I'm a bit unsure/confused now - is it a bug after all? Does it make sense that I continue on the documentation PR for the docker configuration? 

Or will there be a fix and we explain the required settings for docker in the forum until the fix is out?

**Comment by lolodomo** [comment]
@pgfeller : I believe some confusion has unforutnately just been introduced by linking two different "problems" even if both concern (mainly) docker installation. Here we are talking about mDNS and too many listeners being created. @wborn just identified that the reason could be a very recent change I have done (to fix another issue).

Your problem is related to UPnP (so not mDNS) and docker and does not lead to out of memory. That being said, one point of the discussion we have here about mDNS could lead to a good question for the UPnP stuff: your solution is to set some java parameters to set the interface to consider by UPnP; maybe this is something openHAB could set automatically  by itself, considering the network configuration parameters set in openHAB server (in case some are defined) ?

**Comment by SHU-red** [comment]
Thank you very much @wborn --> You helped me out with this

1. I opened https://github.com/openhab/openhab-distro/issues/1619
2. Then i/we thought this could be related to https://github.com/openhab/openhab-addons/issues/15437

But in the end @wborn led me to this topic

Root cause was that i also used openhab in `network_mode: host` which is also suggested in all the prepared `docker_compose` templates

***Result***
- Did remove openHAB from host-mode
- Defined Ports individually
- Re-Deployed / Upgraded to `openHAB 4.1.0 Release Build` successfully --> Normal CPU load and loads of happiness

Thanks to all of you

One last question:
Im not very advanced in this GitHub Release-Management stuff
@wborn is the  PR https://github.com/openhab/openhab-core/pull/3978 considered to be released in future versions and would solve the problem in using host-mode without having to configure something?

**Comment by wborn** [comment]
> Root cause was that i also used openhab in network_mode: host which is also suggested in all the prepared docker_compose templates

It is suggested like that because without using host mode the mDNS/UPnP discovery does not properly work.

>  Normal CPU load and loads of happiness

Nice to see there is a workaround and it is clear what it causing your issues!

> is the PR https://github.com/openhab/openhab-core/pull/3978 considered to be released in future versions and would solve the problem in using host-mode without having to configure something?

The fix was cherry-picked to the 4.1.x branch so it should be part of the next 4.1.1 bugfix release.
